addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/socket/XiaomiDiscoverySocket.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mihome.internal.socket;
+
+import java.io.IOException;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * Takes care of the discovery communication with the MiHome gateway
+ *
+ * @author Dimalo
+ *
+ */
+public class XiaomiDiscoverySocket extends XiaomiSocket {
+
+    private static final int MCAST_PORT = 4321;
+
+    public XiaomiDiscoverySocket() {
+        super();
+    }
+
+    /**
+     * Sets up the {@link XiaomiDiscoverySocket}.
+     *
+     * Connects the socket to the specific multicast address and port.
+     * Starts the {@link ReceiverThread} for the socket.
+     */
+    @Override
+    void setupSocket() {
+        synchronized (XiaomiDiscoverySocket.class) {
+            try {
+                logger.debug("Setup discovery socket");
+                socket = new DatagramSocket(port);
```

Comment:
XiaomiBridgeDiscoveryService -> startScan() -> new XiaomiDiscoverySocket() -> super() -> XiaomiSocket() -> port = 0 -> setupSocket() -> new DatagramSocket(0)

Is that intentional, that DatagramSocket is set to 0? Shouldn't it be 5353 (or 9898, not sure... the port Xiaomi broadcasts to?)